impl/maven-impl/src/main/java/org/apache/maven/impl/DefaultPluginXmlFactory.java

```diff
@@ -62,14 +63,9 @@ public PluginDescriptor read(@Nonnull XmlReaderRequest request) throws XmlReader
                 return xml.read(inputStream, request.isStrict());
             } else if (reader != null) {
                 return xml.read(reader, request.isStrict());
-            } else if (path != null) {
-                try (InputStream is = Files.newInputStream(path)) {
-                    return xml.read(is, request.isStrict());
-                }
-            } else {
-                try (InputStream is = url.openStream()) {
-                    return xml.read(is, request.isStrict());
-                }
+            }
```

Comment:
This throws an NPE if the path is null. I don't think that's what's wanted here

Reply:
see L:56:

`path == null`

This case is not possible. Only in our mind or if JVM is corrupt.

Reply:
early exit then tho IDE makes warning here too, requesting `requireNonNull`.

If path is null like you tell, whats `impossible` imho, then it will exit here again with runtime error.

<img width="799" alt="image" src="https://github.com/user-attachments/assets/27d1e055-d4e2-495f-91a4-2d893a84a9c2" />

Reply:
> If this case really isn't possible

4 params 4 cases thats mandatory.

Now the code has dedicated design making it clear.